server/src/main.rs

```diff
@@ -230,8 +230,12 @@ impl Backend {
         diagnostics.dedup_by(|d1, d2| d1.span() == d2.span() && d1.message() == d2.message());
 
         // Group the diagnostics by file since diagnostics are published per file and diagnostic.span contains the file URL
-        // Process diagnostics and update publish_map
-        process_diagnostics(diagnostics, &mut publish_map);
+        // Process diagnostics and update publish_map. Any diagnostics that do not have a span are returned for further processing.
+        let spanless_diagnostics = process_diagnostics(diagnostics, &mut publish_map);
```

Comment:
Maybe process diagnostics should just show the popup instead of having this logic in two places?

Reply:
I thought about that but then we have to pass the client around a lot more to get it to the `process_diagnostic` call.